src/ExtensionService.re

```diff
@@ -0,0 +1,67 @@
+[@deriving show]
+type msg =
+  | ActivateExtension({
+      extensionId: string,
+      activationEvent: option(string),
+    })
+  | OnWillActivateExtension({extensionId: string})
+  | OnDidActivateExtension({
```

Comment:
The `On...` naming convention here is really odd. It's hard to understand what this means, and even which way it's intended to be go. When this convention is applied to a function, it usually means the callback passed to ti will be called _on_ the given condition. It's a second-order event, in a sense. But there's no callback here, so what does the "on" refer to? Does it mean this will be used somewhere to register a callback that will be called when the event happens? Or does it mean the event actually happened? I would guess the latter, but the amount of mental gymnastics needed to arrive to that conclusion is unnecessarily high. Removing the "on" prefix removes the second-order implication and would therefore make it much less ambiguous.

Reply:
The names come verbatim from the [`exthost.protocol.ts'](https://github.com/onivim/vscode-exthost/blob/923c38b016c87a205957456e13c62f8dfd3bdc62/src/vs/workbench/api/common/extHost.protocol.ts) file, ie:
```
export interface MainThreadExtensionServiceShape extends IDisposable {
	$activateExtension(extensionId: ExtensionIdentifier, reason: ExtensionActivationReason): Promise<void>;
	$onWillActivateExtension(extensionId: ExtensionIdentifier): void;
	$onDidActivateExtension(extensionId: ExtensionIdentifier, codeLoadingTime: number, activateCallTime: number, activateResolvedTime: number, activationReason: ExtensionActivationReason): void;
	$onExtensionActivationError(extensionId: ExtensionIdentifier, error: ExtensionActivationError): Promise<void>;
	$onExtensionRuntimeError(extensionId: ExtensionIdentifier, error: SerializedError): void;
	$onExtensionHostExit(code: number): void;
}
```

I thought that keeping the names in sync with the exthost protocol would be the simplest - but we could try the convention of removing the `on` prefix, if you think that would be clearer.

Reply:
I'll split this out into a separate PR for us to discuss in parallel - that way I can bring this in and continue to unblock other pieces / tests for the API, and we can still track this discussion.

Reply:
Created a separate PR for us to continue discussion here: https://github.com/onivim/reason-vscode-exthost/pull/18